### PR TITLE
support new extension point names format

### PR DESCRIPTION
**Pull request by jonathanhamel4**
### WHY are these changes introduced?

Adds support for both the `Admin::XX::XX` syntax and the `admin.x.x` syntax when extracting the surface of an extension. This will allow to fix the issue where `No extensions found for the surface: admin` is returned in the embedded admin console.

### WHAT is this pull request doing?

Just  changing the regex to support both `::` or `.` as delimiter. (Can current extension points use a `.` that would influence this?)

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: .changeset/neat-avocados-peel.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/app': patch
```
Comment (vividviolet) [inline .changeset/neat-avocados-peel.md:2]: @isaacroldan any concerns with making this a patch?

**Review by vividviolet — COMMENTED** [review]
Thanks for doing this! Just one suggestion - can you add  an extension point with the new naming format to the payload [test](https://github.com/Shopify/cli/blob/6ab492a8f8576befa3a14b113383ce90b32bca2b/packages/app/src/cli/services/dev/extension/payload.test.ts#L136) ? This one is important for local development

File: .changeset/neat-avocados-peel.md (hunk 1 above)
Comment (isaacroldan) [reply .changeset/neat-avocados-peel.md:2]: nop, sounds good 👌

**Review by vividviolet — APPROVED** [review]

**Pull request merged**